Inbound Message Queuing

chrislennon opened the pull request:
Purpose:
In its current state, will relay 'streamer online' messages that are generated into the SQS message queue.

Adds dependencies:
- [sqs-consumer](https://www.npmjs.com/package/sqs-consumer)

Adds environment variables:

| Variable        | Description           | Notes  |
| ------------- |:-------------:| -----:|
| SQS_ACCESS_KEY      | Access key for user with permission to SQS | [IAM](https://console.aws.amazon.com/iam/home?region=eu-west-1) |
| SQS_SECRET_KEY      | Secret key for user with permission to SQS     |   [IAM](https://console.aws.amazon.com/iam/home?region=eu-west-1) |
| SQS_STREAM_QUEUE | URL of queue      |    [SQS](https://eu-west-1.console.aws.amazon.com/sqs/home?region=eu-west-1) |
| STREAM_GUILD      | Guild/Server to report to    |   e.g. Gaymers |
| STREAM_CHANNEL | Channel to report to      |    e.g. notifications |

Usage:

- On start of the bot the sqs-consumer will subscribe to reading of message from the queue url. 
- As messages are received they will output to the server and channel described within environment variables.

The triggering of messages is currently handled by a lambda function which is documented within this [private repository](https://github.com/chrislennon/gaymers-discobot-stream-queue) (reviewers should have access).

chrislennon committed:
adds basic sqs functionality

chrislennon committed:
parameterises the queue and guild

chrislennon committed:
Merge pull request #1 from chrislennon/feature/sqs-messages

Feature/sqs messages

jmariondev commented on index.js:
```diff
@@ -25,6 +25,15 @@ const moment = require('moment');
 const roles = require('./roles');
 
+const Consumer = require('sqs-consumer');
+const AWS = require('aws-sdk');
```
Does this need to be added as a new dep?

chrislennon replied:
Probably not but I kept it there for clarity since the first thing we do is override it's access & secret key. If you are truly offended it can be removed I believe 👼

jmariondev commented on msgq/messageReceived.js:
```diff
@@ -0,0 +1,37 @@
+
+    // Log the user joining to #user-logs
+    if (!notificationChannel) {
+      console.error('Channel #general doesn\'t exist!');
```
Format error string using env var?

chrislennon replied:
[Added](https://github.com/chrislennon/DiscoBot/commit/1c585930e06737c88d1a70e5394c9420b22ed566#diff-12dbd46b5eb6d45fa1161ceac624e12eR31) env var

jmariondev commented on msgq/messageReceived.js:
```diff
@@ -0,0 +1,37 @@
+    const guildObj = bot.guilds.find('name',streamGuild);
+    const notificationChannel = guildObj.channels.find('name', streamChannel);
+
+    // Log the user joining to #user-logs
```
Correctly reflect what this is doing

chrislennon replied:
[Corrected](https://github.com/chrislennon/DiscoBot/commit/1c585930e06737c88d1a70e5394c9420b22ed566#diff-12dbd46b5eb6d45fa1161ceac624e12eR29) comment

jmariondev commented on index.js:
```diff
@@ -127,8 +142,32 @@ events.messageUpdated = require('./events/messageUpdated');
 bot.on('ready', () => {
   console.log('Bot connected to Discord: ', moment(Date.now()).format(format));
+
+const sqsStreamers = Consumer.create({
```
Indents here are wonky?

chrislennon replied:
[UnWonkified](https://github.com/chrislennon/DiscoBot/commit/1c585930e06737c88d1a70e5394c9420b22ed566#diff-168726dbe96b3ce427e7fedce31bb0bcR146)

jmariondev commented on msgq/messageReceived.js:
```diff
@@ -0,0 +1,37 @@
+  process: (bot, message) => {
+    const streamGuild = process.env.STREAM_GUILD;
+    const streamChannel = process.env.STREAM_CHANNEL;
+    const guildObj = bot.guilds.find('name',streamGuild);
```
Cosmetic: space after comma

chrislennon replied:
[OMFGDAFUQ - Fixed](https://github.com/chrislennon/DiscoBot/commit/1c585930e06737c88d1a70e5394c9420b22ed566#diff-12dbd46b5eb6d45fa1161ceac624e12eR26)

chrislennon committed:
format and styling cleanup

as per Lua's comments
https://github.com/gaymers-discord/DiscoBot/pull/229#discussion_r114093204

chrislennon committed:
Merge pull request #2 from chrislennon/feature/sqs-messages

format and styling cleanup
